Add PROFILING_INTERNALS.md.

davidtgoldblatt opened the pull request:
This documents and explains some of the logic behind the profiling implementation.

umanwizard commented:
greatly appreciate making this public!

eminghuliev commented:
![image](https://user-images.githubusercontent.com/14012405/89893204-fbb41480-dbe8-11ea-8c0b-7f878ab7faef.png)
maybe Github isn't able to parse LateX. Did you see that?

davidtgoldblatt commented:
Updated in response to the comments here and on twitter (in particular the thread starting at https://twitter.com/pkhuong/status/1292991562339749889).

It turns out that github markdown won't render math at all. I just added a note to say "hey should you should probably run pandoc on this".
